static/js/components/ClassificationForm.jsx

```diff
@@ -0,0 +1,207 @@
+    probability_errored: false,
+    allowed_classes: latest_taxonomyList.length > 0 ?
+      latest_taxonomyList[0].allowed_classes : [null]
+  };
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const { handleSubmit } = useForm();
+
+  const useStyles = makeStyles((theme) => ({
+    formControl: {
+      margin: theme.spacing(1),
+      fullWidth: true,
+      display: 'flex',
+      wrap: 'nowrap'
+    }
+  }));
+  const classes = useStyles();
+
+  if (latest_taxonomyList.length === 0) {
+    return (
+      <b>
+        No taxonomies loaded...
+      </b>
+    );
+  }
+
+  const handleTaxonomyChange = (event) => {
+    dispatch({ name: "taxonomy_index", value: event.target.value });
+    dispatch({ name: "classification", value: "" });
+    dispatch({ name: "class_select_enabled", value: true });
+    dispatch({ name: "probability_select_enabled", value: false });
+    dispatch({ name: "probability_errored", value: false });
+    dispatch({ name: "probability", value: 1.0 });
+  };
+
+  const handleClasschange = (event, value) => {
+    dispatch({ name: "classification", value });
+    dispatch({ name: "probability_select_enabled", value: true });
+  };
+
+  const processProb = (event) => {
+    // make sure that the probability in in [0,1], otherwise set
+    // an error state on the entry
+    if ((Number.isNaN(parseFloat(event.target.value))) ||
+       ((parseFloat(event.target.value) > 1) ||
+       (parseFloat(event.target.value) < 0))) {
+      dispatch({ name: "probability_errored", value: true });
+    } else {
+      dispatch({ name: "probability_errored", value: false });
+      dispatch({ name: "probability", value: event.target.value });
+    }
+  };
+
+  const onSubmit = () => {
+    // TODO: allow fine-grained user groups in submission
+    const formData = {
+      taxonomy_id: latest_taxonomyList[state.taxonomy_index].id,
+      obj_id,
+      classification: state.classification.class,
+      probability: parseFloat(state.probability)
+    };
+    // We need to include this field in request, but it isn't in form data
+    submitDispatch(Actions.addClassification(formData));
+  };
+
+
+  return (
+    <div>
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <div>
+          <h3>Add Classification</h3>
+          <FormControl className={classes.formControl}>
+            <InputLabel id="taxonomy-label">Taxonomy</InputLabel>
+            <Select
+              id="tax-select"
+              defaultValue=""
+              onChange={handleTaxonomyChange}
+            >
+              {latest_taxonomyList.map((taxonomy, index) => makeMenuItem(
+                taxonomy, index
+              ))}
+            </Select>
+          </FormControl>
+          <div style={{ display: state.class_select_enabled ? "block" : "none" }}>
+            <Autocomplete
+              options={state.allowed_classes}
+              id="classification"
+              getOptionSelected={(option) => {
```

Comment:
how does this ever return anything other than `true`?

Reply:
Fixed to be more explicit.